planetscale/imports.go

```diff
@@ -0,0 +1,321 @@
+	DataImportSwitchTrafficRunning
+	DataImportSwitchTrafficCompleted
+	DataImportSwitchTrafficError
+	DataImportReverseTrafficRunning
+	DataImportReverseTrafficCompleted
+	DataImportReverseTrafficError
+	DataImportDetachExternalDatabaseRunning
+	DataImportDetachExternalDatabaseError
+	DataImportReady
+)
+
+func (d DataImportState) String() string {
+	switch d {
+	case DataImportPreparingDataCopy:
+		return "preparing to copy data from external database"
+	case DataImportPreparingDataCopyFailed:
+		return "failed to copy data from external database"
+	case DataImportCopyingData:
+		return "copying data from external database"
+	case DataImportCopyingDataFailed:
+		return "failed to copy data from external database"
+	case DataImportSwitchTrafficPending:
+		return "PlanetScale database is running in replica mode"
+	case DataImportSwitchTrafficRunning:
+		return "switching PlanetScale database to primary mode"
+	case DataImportSwitchTrafficError:
+		return "failed to switching PlanetScale database to primary mode"
+	case DataImportReverseTrafficRunning:
+		return "switching PlanetScale database to replica mode"
+	case DataImportReverseTrafficError:
+		return "failed to switching PlanetScale database to replica mode"
+	case DataImportDetachExternalDatabaseRunning:
+		return "detaching external database from  PlanetScale database"
+	case DataImportDetachExternalDatabaseError:
+		return "failed to detach external database from  PlanetScale database"
+	}
+
+	return "all done"
+}
+
+type DataImport struct {
+	ID          string `json:"id"`
+	ImportState DataImportState
+	State       string    `json:"state"`
+	Errors      string    `json:"import_check_errors"`
+	StartedAt   time.Time `json:"started_at"`
+	FinishedAt  time.Time `json:"finished_at"`
+	DeletedAt   time.Time `json:"deleted_at"`
+}
+
+func (di *DataImport) ParseState() {
+	switch di.State {
+	case "prepare_data_copy_pending":
+		di.ImportState = DataImportPreparingDataCopy
+	case "prepare_data_copy_error":
+		di.ImportState = DataImportPreparingDataCopyFailed
+	case "data_copy_pending":
+		di.ImportState = DataImportCopyingData
+	case "data_copy_error":
+		di.ImportState = DataImportCopyingDataFailed
+	case "switch_traffic_workflow_pending":
+		di.ImportState = DataImportSwitchTrafficPending
+	case "switch_traffic_workflow_running":
+		di.ImportState = DataImportSwitchTrafficRunning
+	case "switch_traffic_workflow_error":
+		di.ImportState = DataImportSwitchTrafficError
+	case "reverse_traffic_workflow_running":
+		di.ImportState = DataImportReverseTrafficRunning
+	case "reverse_traffic_workflow_error":
+		di.ImportState = DataImportReverseTrafficError
+	case "cleanup_workflow_pending":
+		di.ImportState = DataImportSwitchTrafficCompleted
+	case "cleanup_workflow_running":
+		di.ImportState = DataImportDetachExternalDatabaseRunning
+	case "cleanup_workflow_error":
+		di.ImportState = DataImportDetachExternalDatabaseError
+	default:
+		di.ImportState = DataImportReady
```

Comment:
If this is what we want to default to, should it be the default state for the `iota` definition we have above? This way when unspecified, it won't default to `DataImportPreparingDataCopy`.

Reply:
💯 , The default should be `DataImportPreparingDataCopy`, I'll fix it.